### PR TITLE
Add entity metadata extractor

### PR DESCRIPTION
**Pull request by logan-markewich**
# Description

This uses the a new multilingual metadata extractor to extract entities. The original model card is [here](https://huggingface.co/tomaarsen/span-marker-xlm-roberta-base-multinerd).

## Type of Change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] Add extractor
- [x] Add notebook
- [x] Update docs

### CODE REVIEW

**Comment by logan-markewich** [comment]
Example output from PG essay
image

File: llama_index/node_parser/extractors/metadata_extractors.py [hunk 1]
```diff
@@ -332,3 +332,88 @@ def extract(self, nodes: Sequence[BaseNode]) -> List[Dict]:
...
+
+        self._model = SpanMarkerModel.from_pretrained(model_name)
+        if device is not None:
+            self._model = self._model.to(device)
+
+        self._tokenizer = tokenizer or word_tokenize
+        self._prediction_threshold = prediction_threshold
+        self._span_joiner = span_joiner
+        self._label_entities = label_entities
+        self._entity_map = DEFAULT_ENTITY_MAP
```
Comment (jerryjliu) [inline llama_index/node_parser/extractors/metadata_extractors.py:398]: nit: entity_map arg isn't used

Comment (jerryjliu) [reply llama_index/node_parser/extractors/metadata_extractors.py:398]: also might be good to briefly add to docstring

**Review by jerryjliu — APPROVED** [review]
nice - some minor nits but otherwise looks good

File: llama_index/node_parser/extractors/metadata_extractors.py (hunk 1 above)
Comment (logan-markewich) [reply llama_index/node_parser/extractors/metadata_extractors.py:398]: will do!

Comment (logan-markewich) [reply llama_index/node_parser/extractors/metadata_extractors.py:398]: (entity map is actually used in the line below, as well as in the extract function 👍🏻)

Comment (logan-markewich) [reply llama_index/node_parser/extractors/metadata_extractors.py:418]: docstring added!

**Pull request merged**